mp/src/vgui2/vgui_controls/CvarTextEntry.cpp

```diff
@@ -62,6 +62,23 @@ void CvarTextEntry::GetSettings(KeyValues* pOutResource)
     pOutResource->SetString("cvar_name", m_cvarRef.GetName());
 }
 
+void CvarTextEntry::SetText(const char* text)
+{
+    if (text[strnlen(text, MAX_CVAR_TEXT) - 1] == '.')
+    {
+        // trying to enter a decimal number so set the text normally
+        BaseClass::SetText(text);
+    }
+    else // not in the middle of setting a decimal number
+    {
+        // making sure the formatting is correct (removing trailing zeros via %g)
+        char newText[MAX_CVAR_TEXT];
+        Q_strncpy(newText, text, MAX_CVAR_TEXT);
```

Comment:
Is there a reason for doing the copy?

Reply:
Just that the incoming string is `const`. Could remove that, using an overload rather than an override I suppose.

Reply:
Is it possible to remove this and only do the snprintf?

Reply:
Woops! My Brain small

I removed the strncpy. Have to keep the new text declaration though as the incoming string is const. Can move this to another function that isn't a `SetText` override instead.

Reply:
`newText` is fine!